### ISSUE TITLE
Ability to show nominator names on content

### ISSUE THREAD

**Issue by boonebgorges** [PressForward/pressforward#1104]
It should be possible to display the name of the nominating user on a promoted post.

A few notes on the desired functionality:

1. It probably makes sense to have this information appear near the source statement. But we should provide template functions as necessary so that custom themes can place the attribution where they'd like.
2. I imagine we'd have the feature turned off by default. There'd be a way to turn it on in code (most likely a filter) and likely also a toggle in the UI
3. We should use the 'display_name' of the contributing user.

**Comment by boonebgorges** [comment]
Thinking more about this, I think it probably makes sense to build a 'Nominators' block. This gives editors the ability to place the nominator info wherever they'd like, and it would nicely parallel other suggestions like #1052.

The problem with this is default behavior. Some sites will want this block to appear by default, while others don't. So we'll need a technique for inserting the block into the post content, probably at the time that a nomination is promoted to draft. Let's tackle this alongside other block-editor focused items, such as #1052 and #1124.

**Comment by boonebgorges** [comment]
I've got a working version of this block. Here's a gif that demonstrates how it's inserted into a post and then customized.

![Peek 2024-01-02 15-34](https://github.com/PressForward/pressforward/assets/246627/b2b4cd5f-6671-4252-9021-763c59b8636f)

A few features of note:

- The 'Nominated by' text is customizable. The nominator names are generated dynamically.
- I built the markup in such a way that it inherits as much styling as possible from the theme, and also permits various sorts of visual design using CSS. (It's two paragraph tags nested in the block div.) In this initial version, I've added 'Inline' and 'Stacked' styles in the interface.
- The block supports WP's tools for background color, font color, font size. As a default, I've given the item a very light-grey background with a bit of padding, and a font-size of .9em. This is enough to set it apart, and generic enough that it will probably look OK in most themes. This can be customized on a given site by either (a) overriding styles in a stylesheet, (b) manually modifying the styles using the interface in the Block Editor whenever adding the block to a post, or (c) adding global block defaults either in your theme.json or using the Styles tool https://wpblocktips.com/tip/how-to-set-default-styles-for-blocks/
- The block doesn't appear on the front end if the item has no nominators (for whatever reason)

Because this feature is built using a custom block, a site that is running a block theme and wants to have a Nominators section on each published Post (without having to add the block manually each time) by adding the block to the Single Post template in the Site Editor.
